.github/workflows/test.yml

```diff
@@ -0,0 +1,205 @@
+          echo "holdings-date=$HOLDINGS_DATE"
+          echo "holdings-date=$HOLDINGS_DATE" >> "$GITHUB_OUTPUT"
+
+          PACTA_DATA_URL="$(jq -rc '.pactaDataURL' $config_file)"
+          echo "pacta-data-url=$PACTA_DATA_URL"
+          echo "pacta-data-url=$PACTA_DATA_URL" >> "$GITHUB_OUTPUT"
+
+          BENCHMARKS_URL="$(jq -rc '.benchmarksURL' $config_file)"
+          echo "benchmarks-url=$BENCHMARKS_URL"
+          echo "benchmarks-url=$BENCHMARKS_URL" >> "$GITHUB_OUTPUT"
+
+          RESULTS_URL="$(jq -rc '.resultsURL' $config_file)"
+          echo "results-url=$RESULTS_URL"
+          echo "results-url=$RESULTS_URL" >> "$GITHUB_OUTPUT"
+
+          # includes handling for null/missing keys
+          PARAMETERS="$(jq -rc '.parameters | select( . != null )' $config_file)"
+          echo "parameters=$PARAMETERS"
+          echo "parameters=$PARAMETERS" >> "$GITHUB_OUTPUT"
+
+      # https://github.com/Azure/login?tab=readme-ov-file#login-with-openid-connect-oidc-recommended
+      - name: Azure Login
+        uses: azure/login@v2
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      # https://github.com/marketplace/actions/azure-cli-action#workflow-to-execute-an-azure-cli-script-of-a-specific-cli-version
+      - name: Download outputs from workflow.pacta
+        uses: azure/CLI@v2
+        env:
+          RESULTS_URL: ${{ steps.prepare.outputs.results-url }}
+        with:
+          # azcliversion: 2.30.0
+          inlineScript: |
+            az storage copy \
+              --source "$RESULTS_URL/*" \
+              --destination "analysis_output_dir/" \
+              --recursive \
+              --exclude-pattern "*.sqlite"
+
+      # https://github.com/marketplace/actions/azure-cli-action#workflow-to-execute-an-azure-cli-script-of-a-specific-cli-version
+      - name: Download Benchmark Results
+        uses: azure/CLI@v2
+        env:
+          BENCHMARKS_URL: ${{ steps.prepare.outputs.benchmarks-url }}
+        with:
+          # azcliversion: 2.30.0
+          inlineScript: |
+            az storage copy \
+              --source "$BENCHMARKS_URL/*" \
+              --destination "benchmarks_dir/" \
+              --recursive \
+              --exclude-pattern "*.sqlite"
+
+      - name: Run Docker Image
+        env:
+          FULL_IMAGE_NAME: ${{ inputs.full-image-name }}
+          ANALYSIS_OUTPUT_DIR: analysis_output_dir
+          BENCHMARKS_DIR: benchmarks_dir
+          REPORT_OUTPUT_DIR: report_output_dir
+          SUMMARY_OUTPUT_DIR: summary_output_dir
+          WORKSPACE: ${{ github.workspace }}
+          PARAMETERS: ${{ steps.prepare.outputs.parameters }}
+          REAL_ESTATE_DIR: real_estate_dir
+          SURVEY_DIR: survey_dir
+        run: |
+          mkdir -p "$REPORT_OUTPUT_DIR"
+          chmod -R 777 "$REPORT_OUTPUT_DIR"
```

Comment:
nit: still thing there must be a safer solution than 777

Reply:
Agree, but don't know what that is yet. dealing with the permissions in the bind mounts is tricky.